tools/data-api/internal/repositories/services.go

```diff
@@ -1,227 +1,411 @@
 package repositories
 
 import (
+	"encoding/json"
 	"fmt"
-	"net/http"
-
-	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"os"
+	"strings"
 )
 
 type ServicesRepository interface {
 	GetByName(serviceName string, serviceType ServiceType) (*ServiceDetails, error)
-
 	GetAll(serviceType ServiceType) (*[]ServiceDetails, error)
+	ClearCache() error
 }
 
 var _ ServicesRepository = &ServicesRepositoryImpl{}
 
 type ServicesRepositoryImpl struct {
-	// TODO:
-	// directory string
+	// directory is where the api definitions for all services exist - this is where the server will read from
+	directory string
+
+	// graphV1 is a map of service name to string
+	graphV1 *map[string]string
+
+	// graphV2 is a map of service name to string
+	graphV2 *map[string]string
+
+	// resourceManager is map of service name to ServiceDetails, the ServiceDetails contain all the
+	// Service, Version and Resource definitions loaded and unmarshalled from the JSON API definitions
+	resourceManager *map[string]ServiceDetails
```

Comment:
since each `api` gets it's own repository, we're going to have 3 separate instances of the `repository`, which should all be using the same format on-disk, so this can be reduced to:

```suggestion
	// services is a  map of Service Name to ServiceDetails
	// ServiceDetails contains all the Service, Version and Resource definitions which
	// are loaded and unmarshalled from the JSON API definitions
	services *map[string]ServiceDetails
```